pkg/cel/policies/dpol/engine/fetch.go

```diff
@@ -13,31 +13,42 @@ import (
 type fetchProvider struct {
 	compiler     compiler.Compiler
 	dpolLister   policiesv1alpha1listers.DeletingPolicyLister
+	ndpolLister  policiesv1alpha1listers.NamespacedDeletingPolicyLister
 	polexLister  policiesv1alpha1listers.PolicyExceptionLister
 	polexEnabled bool
 }
 
 func NewFetchProvider(
 	compiler compiler.Compiler,
 	dpolLister policiesv1alpha1listers.DeletingPolicyLister,
+	ndpolLister policiesv1alpha1listers.NamespacedDeletingPolicyLister,
 	polexLister policiesv1alpha1listers.PolicyExceptionLister,
 	polexEnabled bool,
 ) *fetchProvider {
 	return &fetchProvider{
 		compiler:     compiler,
 		dpolLister:   dpolLister,
+		ndpolLister:  ndpolLister,
 		polexLister:  polexLister,
 		polexEnabled: polexEnabled,
 	}
 }
 
-func (r *fetchProvider) Get(ctx context.Context, name string) (Policy, error) {
-	policy, err := r.dpolLister.Get(name)
+func (r *fetchProvider) Get(ctx context.Context, namespace, name string) (Policy, error) {
+	var (
+		policy policiesv1alpha1.DeletingPolicyLike
+		err    error
+	)
+	if namespace == "" {
+		policy, err = r.dpolLister.Get(name)
+	} else {
+		policy, err = r.ndpolLister.NamespacedDeletingPolicies(namespace).Get(name)
+	}
 	if err != nil {
 		return Policy{}, err
 	}
 	if policy == nil {
-		return Policy{}, fmt.Errorf("deleting policy %s not found", name)
+		return Policy{}, fmt.Errorf("deleting policy %s/%s not found", namespace, name)
```

Comment:
We should adapt the error to the type of policy (only use the name if it is a clustered policy)